README.md
# HoverX

[![Pub Version](https://img.shields.io/pub/v/badges?color=blueviolet)](https://pub.dev/packages/badges) [![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat)](https://github.com/dwyl/esta/issues) [![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://GitHub.com/Naereen/StrapDown.js/graphs/commit-activity) ![Maintaner](https://img.shields.io/badge/maintainer-theMaintainer-blue) ![Profile views](https://gpvc.arturio.dev/Brewnbeer) 

<img alt="Dart" src="https://img.shields.io/badge/dart-%230175C2.svg?style=for-the-badge&logo=dart&logoColor=white"/>  <img alt="Flutter" src="https://img.shields.io/badge/Flutter-%2302569B.svg?style=for-the-badge&logo=Flutter&logoColor=white" />

Hover Text over Image using HoverX

Using HoverX you can achieve beautiful results on any website with little effort. Hover effects are probably the most used elements in web design, mainly because of the ease of implementing them coupled with a greatly improved user experience. HoverX create an opportunity to add interactivity to elements on a website without slowing it down.

## Demo

[![ForTheBadge built-with-love](http://ForTheBadge.com/images/badges/built-with-love.svg)](https://GitHub.com/Naereen/) 

```dart
import 'package:flutter/material.dart';
import 'package:hoverx/hoverx.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: Center(
        child: HoverX(
          title: "Hello World",
          image: NetworkImage(
              "https://images.unsplash.com/32/Mc8kW4x9Q3aRR3RkP5Im_IMG_4417.jpg?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1950&q=80"),
          hoverColor: Colors.black,
        ),
      ),
    );
  }
}
```


CHANGELOG.md
## [1.0.2] - Added Example

* Added Example

## [1.0.1] - Migrating to Null Safety.

* Migrated to Null Safety.
* ChangeLog Updates

## [1.0.0] - Migrating to Null Safety.

* Migrated to Null Safety.

## [0.0.1] - 20-05-2021.

* Initial release.





test/hover_effect_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:hover_effect/hover_effect.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
  });
}


lib/hoverx.dart
library hoverx;

import 'package:flutter/material.dart';

class HoverX extends StatefulWidget {
  final String title;
  final TextStyle? titleStyle;
  final ImageProvider image;
  final Color hoverColor;
  final double height;
  final double? width;
  final BorderRadiusGeometry? borderRadius;

  const HoverX({
    Key? key,
    required this.title,
    this.titleStyle,
    required this.image,
    required this.hoverColor,
    this.height = 370,
    this.width,
    this.borderRadius,
  }) : super(key: key);

  @override
  _HoverXState createState() => _HoverXState();
}

class _HoverXState extends State<HoverX> {
  bool isVisible = false;

  @override
  Widget build(BuildContext context) {
    return MouseRegion(
      onEnter: (f) {
        setState(() {
          isVisible = true;
        });
      },
      onExit: (f) {
        setState(() {
          isVisible = false;
        });
      },
      child: Container(
        width: widget.width ?? MediaQuery.of(context).size.width,
        height: widget.height,
        decoration: BoxDecoration(
          color: widget.hoverColor,
          borderRadius: widget.borderRadius ?? BorderRadius.circular(0),
          image: DecorationImage(
            image: widget.image,
          ),
        ),
        child: Visibility(
          visible: isVisible,
          child: Container(
            width: widget.width ?? MediaQuery.of(context).size.width,
            height: widget.height,
            color: Colors.white70,
            child: Center(
              child: Text(
                widget.title,
                style:
                    widget.titleStyle ?? Theme.of(context).textTheme.headline3,
              ),
            ),
          ),
        ),
      ),
    );
  }
}


lib/hover_effect.dart
library hoverx;

import 'package:flutter/material.dart';

class HoverEffect extends StatefulWidget {
  final String title;
  final TextStyle titleStyle;
  final ImageProvider image;
  final Color hoverColor;
  final double height;
  final double width;
  final BorderRadiusGeometry borderRadius;

  const HoverEffect({
    Key key,
    @required this.title,
    this.titleStyle,
    @required this.image,
    @required this.hoverColor,
    this.height = 370,
    this.width,
    this.borderRadius,
  }) : super(key: key);

  @override
  _HoverEffectState createState() => _HoverEffectState();
}

class _HoverEffectState extends State<HoverEffect> {
  bool isVisible = false;

  @override
  Widget build(BuildContext context) {
    return MouseRegion(
      onEnter: (f) {
        setState(() {
          isVisible = true;
        });
      },
      onExit: (f) {
        setState(() {
          isVisible = false;
        });
      },
      child: Container(
        width: widget.width ?? MediaQuery.of(context).size.width,
        height: widget.height,
        decoration: BoxDecoration(
          color: widget.hoverColor,
          borderRadius: widget.borderRadius ?? BorderRadius.circular(0),
          image: DecorationImage(
            image: widget.image,
          ),
        ),
        child: Visibility(
          visible: isVisible,
          child: Container(
            width: widget.width ?? MediaQuery.of(context).size.width,
            height: widget.height,
            color: Colors.white70,
            child: Center(
              child: Text(
                widget.title,
                style:
                    widget.titleStyle ?? Theme.of(context).textTheme.headline3,
              ),
            ),
          ),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:hoverx/hoverx.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: Center(
        child: HoverX(
          title: "Hello World",
          image: NetworkImage(
              "https://images.unsplash.com/32/Mc8kW4x9Q3aRR3RkP5Im_IMG_4417.jpg?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1950&q=80"),
          hoverColor: Colors.black,
        ),
      ),
    );
  }
}
